### PR TITLE
Fix pipeline backpressure to exclude treating committed block as pending

### PR DESCRIPTION
**Pull request by igor-aptos**
* Fix pipeline backpressure to exclude treating committed block as pending

For pipeline backpressure we were looking at latency between ordered and committed block.
But committed block is not in the pipeline any more, and next block after it could've
been created significantly later (and so got into the pipeline significantly later).

Second it is better to take actual time block is in the pipeline, than  use ordered timestamp.

if it took us 1s to order oldest_not_committed_block, but 100ms to order
currently ordered block, we don’t want to backpressure if it
is 1.5s passed the oldest_not_committed_block - because it was really only 500ms in the pipeline

### CODE REVIEW

File: consensus/src/block_storage/block_store.rs [hunk 1]
```diff
@@ -482,11 +482,16 @@ impl BlockStore {
...
-        let committed_timestamp = Duration::from_micros(self.commit_root().timestamp_usecs());
+        let ordered_root = self.ordered_root();
+        let commit_root = self.commit_root();
+        let pending_path = self.path_from_commit_root(self.ordered_root().id()).unwrap_or_default();
+
+        let ordered_round = ordered_root.round();
+        let commit_round = commit_root.round();
+        let pending_rounds = pending_path.len();
+        let ordered_timestamp = Duration::from_micros(ordered_root.timestamp_usecs());
+        let oldest_not_committed_timestamp = pending_path.last().map(|b| Duration::from_micros(b.timestamp_usecs())).unwrap_or(Duration::ZERO);
```
Comment (zekun000) [inline consensus/src/block_storage/block_store.rs:493]: are we sure about the order here? I see the code here https://github.com/aptos-labs/aptos-core/blob/88c987c2d14a8fbfffac004192b8304b2871024d/consensus/src/block_storage/block_tree.rs#L394 doesn't match the comments https://github.com/aptos-labs/aptos-core/blob/7a5903a52dffd5595c34b29c671a571a46dd1408/consensus/src/block_storage/mod.rs#L36

Comment (igor-aptos) [reply consensus/src/block_storage/block_store.rs:493]: should I depend on order, or just take min?

Comment (zekun000) [reply consensus/src/block_storage/block_store.rs]: min seems a safer choice 😄 iterating through this should be negligible

**Comment by sitalkedia** [comment]
@igor-aptos - A broader question - do we need a pipeline backpressure on consensus if the quorum store backpressure works as expected?

**Comment by igor-aptos** [comment]
In general,  this is back-pressure from execution/pipeline being too slow, on the upstream.

QS currently has no info from execution,  and overall it seems cleaner to me to have backpressure directly on the component that is overloading another component. 

But if you have a concrete suggestion- we can discuss

**Comment by igor-aptos** [comment]
Updated PR to further improve the logic - to use time oldest not committed block has spent in the pipeline.

if it took us 1s to order oldest_not_committed_block, but 100ms to order
currently ordered block, we don’t want to backpressure if it
is 1.5s passed the oldest_not_committed_block - because it was really only 500ms in the pipeline

File: testsuite/smoke-test/src/aptos_cli/validator.rs [hunk 2]
```diff
@@ -198,6 +198,7 @@ async fn test_onchain_config_change() {
             let inner = match genesis_config.consensus_config.clone() {
                 OnChainConsensusConfig::V1(inner) => inner,
                 OnChainConsensusConfig::V2(inner) => inner,
+                OnChainConsensusConfig::V3(inner) => inner.main,
```
Comment (bchocho) [inline testsuite/smoke-test/src/aptos_cli/validator.rs:201]: unrelated change

**Review by bchocho — APPROVED** [review]
LGTM. Two comments below

File: testsuite/smoke-test/src/aptos_cli/validator.rs (hunk 2 above)
Comment (igor-aptos) [reply testsuite/smoke-test/src/aptos_cli/validator.rs:201]: it's kinda not, I was running this test to see the logs, and the test is broken (it's ignore), so I fixed it here :)

File: consensus/src/block_storage/block_store.rs [hunk 3]
```diff
@@ -500,24 +517,40 @@ impl BlockStore {
...
-        latency_to_committed
-            .saturating_sub(latency_to_ordered.min(MAX_ORDERING_PIPELINE_LATENCY_REDUCTION))
+        counters::CONSENSUS_PROPOSAL_PENDING_DURATION
+            .observe(oldest_not_committed_spent_in_pipeline.as_secs_f64());
+
+        if pending_rounds > 1 {
+            // TODO cleanup
+            // previous logic was using difference between committed and ordered.
+            // keeping it until we test out the new logic.
+            // latency_to_oldest_not_committed
```
[the remark this answers is not in the document]
Comment (igor-aptos) [reply consensus/src/block_storage/block_store.rs:547]: I wanted to leave it here, if we wanted to revert the new change. but if we want to do that - we should revert it to the fix, not before the fix. :)

**Review by zekun000 — APPROVED** [review]
